### PR TITLE
Add adopters.md

### PR DESCRIPTION
**Pull request by SamBarker**
### Type of change

- Enhancement / new feature
- Documentation

### Description

Adds an adopters file so users can signal their use of project.

### Additional Context

So we, anyone else can get a sense of the sort of workloads and situations the project is being used in.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonar-Lint warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: ADOPTERS.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Kroxylicious adopters
```
Comment (tombentley) [inline ADOPTERS.md:1]: Isn't the file name more conventionally `ADOPTERS.md`

File: ADOPTERS.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------|
+| [![Red Hat](https://kroxylicious.io/assets/images/RedHatDark.svg#gh-dark-mode-only)](https://access.redhat.com/products/red-hat-amq-streams/#gh-dark-mode-only)[![Red Hat](https://kroxylicious.io/assets/images/RedHatLight.svg#gh-light-mode-only)](https://access.redhat.com/products/red-hat-amq-streams/#gh-light-mode-only) | AMQ Streams 2.7+   |
+
+
+If you are using Kroxylicious in any context: production, build environment or for developer workstations. Please feel free to open a pull request against this file adding your org and optionally what your using Kroxylicious for.
```
Comment (tombentley) [inline ADOPTERS.md:8]:
```suggestion
If you are using Kroxylicious in any context: production, build environment or for developer workstations, please feel free to open a pull request against this file adding your org and optionally what your using Kroxylicious for.
```

File: ADOPTERS.md (hunk 1 above)
Comment (SamBarker) [reply ADOPTERS.md:1]: You have me slightly confused with this one, isn't that what I called the file? I fully expect I could have typo'd it but your suggestion is the same as what I have.

Comment (SamBarker) [reply ADOPTERS.md:1]: I did get it wrong in the branch name, but the file in the PR is right.

File: ADOPTERS.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------|
+| [![Red Hat](https://kroxylicious.io/assets/images/RedHatDark.svg#gh-dark-mode-only)](https://access.redhat.com/products/red-hat-amq-streams/#gh-dark-mode-only)[![Red Hat](https://kroxylicious.io/assets/images/RedHatLight.svg#gh-light-mode-only)](https://access.redhat.com/products/red-hat-amq-streams/#gh-light-mode-only) | AMQ Streams 2.7+   |
+
+
+If you are using Kroxylicious in any context: production, build environment or for developer workstations, please feel free to open a pull request against this file adding your org and optionally what your using Kroxylicious for.
```
Comment (k-wall) [inline ADOPTERS.md:8]: Maybe move this sentence above the table.

**Review by k-wall — APPROVED** [review]

**Review by k-wall — APPROVED** [review]

**Pull request merged**